web-marketplace/src/components/organisms/PaymentInfoForm/PaymentInfoForm.tsx

```diff
@@ -148,62 +168,108 @@ export const PaymentInfoForm = ({
     );
   };
 
-  return (
-    <Form
-      form={form}
-      onSubmit={async (values: PaymentInfoFormSchemaType) => {
-        setCardDetailsMissing(false);
-        const card = paymentOption === PAYMENT_OPTIONS.CARD;
-        if (card && !stripe) {
-          return;
-        }
-        if (card && stripe && elements && !values.paymentMethodId) {
-          const submitRes = await elements.submit();
-          if (submitRes?.error?.message) {
-            setError(submitRes?.error?.message);
+  const handleOnSubmit = useCallback(
+    async (values: PaymentInfoFormSchemaType) => {
+      // If a logged in user with no email address (web3 account) provides one,
+      // we send a confirmation email.
+      if (
+        !!activeAccount &&
+        !privActiveAccount?.email &&
+        data?.email &&
+        token
+      ) {
+        try {
+          const response = await postData({
+            url: `${apiUri}/marketplace/v1/auth/email/create-token`,
+            data: { email: data?.email },
+            token,
+            retryCsrfRequest,
+          });
+          // If the email is already connected to another account, we show a modal
+          // with the error message.
+          if (response.error === CONNECTED_EMAIL_ERROR) {
+            setConnectedEmailErrorModalAtom(atom => {
+              atom.open = true;
+              atom.email = data?.email!;
+            });
             return;
           }
-          // Create the ConfirmationToken using the details collected by the Payment Element
-          if (values.savePaymentMethod)
-            elements.update({ setupFutureUsage: 'on_session' });
-          const { error, confirmationToken } =
-            await stripe.createConfirmationToken({
-              elements,
-              params: {
-                payment_method_data: {
-                  billing_details: {
-                    name: values.name,
-                    email: values.email,
-                  },
+        } catch (e) {
+          setErrorBannerTextAtom(String(e));
+        }
+      }
+
+      setCardDetailsMissing(false);
+      const card = paymentOption === PAYMENT_OPTIONS.CARD;
+
+      if (card && !stripe) return;
```

Comment:
Should we add feedback for the users here?

```suggestion
      if (card && !stripe) {
        setError('Stripe instance is not available. Please try again later.');
        return;
      }
```

Reply:
@blushi @erikalogie what do you think about adding this?